week3/homework/src/functions.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+const express = require('express');
+
+const uuid = require('uuid/v4');
+
+const app = express();
+
+const {
+  readFile: _readFile,
+  writeFile: _writeFile
+} = require('fs');
+
+const {
+  promisify
+} = require('util');
+
+const readFile = promisify(_readFile);
+const writeFile = promisify(_writeFile);
+
+
+
+//USE BUILT-IN JSON middleware to automatically parse JSON
+app.use(express.json());
+
+const TODO_FILE = 'todo.json';
+
+function readTodos() {
+  return readFile(TODO_FILE, 'utf-8').then(JSON.parse, () => []);
+}
+
+function writeTodos(todos) {
+  return writeFile(TODO_FILE, JSON.stringify(todos));
+}
+
+//Find Todo
+async function findTodo(req) {
+  const id = req.params.id;
+  const todos = await readTodos();
+  const todo = todos.find(todos => todos.id === id);
+  const done = req.body.done;
+  return await [
+    id,
+    req.body,
+    done,
+    todo,
+    todos
+  ];
+}
+
+//MarkTodo Function
+async function markTodo(req, res, doneFlag) {
+  const [, ,
+    done,
```

Comment:
You're not using this variable anywhere.

Reply:
when i try to delete ( , , done , ) i get error !!

Reply:
Your `findTodo` should only return what you need and are going to use. So `findTodo` should only return a `todo` and the `todos` (I don't think you are using `id`, `req.body` and `done` anywhere).